### PR TITLE
Stop green runs from reporting two errors that are not there

### PR DESCRIPTION
**Pull request by johanhaleby**
Every green run reported 2 errors under Annotations, one per JDK, both from the `mongodb-native` shard with the message `Expected`. Nothing was actually failing, which is the problem, since a run that always claims 2 errors trains you to ignore the annotations and a real one then goes unnoticed.

The source is `NativeMongoCheckpointStorageTest`, in the test that disables retry and makes the consumer throw on the first event. With `RetryStrategy.none()` there is nothing left to restart the dispatcher thread, so the exception reaches the JVM's default handler, which prints the stack trace, and GitHub Actions turns that into an error annotation.

I left the production code alone. `NativeMongoSubscriptionModel` logs "Will restart!" and rethrows, and `startSubscription` wraps the runnable in `executeWithRetry`, so with a configured retry strategy it really does restart. With retry explicitly off, letting the exception propagate is the only thing it can do. The test already called Awaitility's `dontCatchUncaughtExceptions`, so the escaping exception was known and intended when it was written.

Instead the test installs a default uncaught exception handler that ignores the one `IllegalArgumentException` it asks for and delegates everything else to whatever handler was installed before, restored after `shutdown()` in the existing `@AfterEach` rather than a second one, since JUnit does not promise an order between two. The injected failure and the assertions are untouched, and the model still logs its warning, so the error path the test exercises still runs.

Checked locally against the same module both ways. `Exception in thread` appears once before this change and not at all after, with 13 tests passing either way. On CI the run is green with zero error annotations, against 2 on every green run before.

### CODE REVIEW

File: subscription/mongodb/native/blocking-checkpoint-storage/src/test/java/org/occurrent/subscription/mongodb/nativedriver/blocking/NativeMongoCheckpointStorageTest.java [hunk 1]
```diff
@@ -123,6 +125,33 @@ void create_mongo_event_store() {
...
         subscriptionModel.shutdown();
         mongoClient.close();
+        // Restored after the shutdown above rather than from a second @AfterEach, because JUnit does not promise
+        // an order between two of them and shutting down must still be covered by the handler.
+        if (replacedDefaultUncaughtExceptionHandler) {
+            Thread.setDefaultUncaughtExceptionHandler(previousDefaultUncaughtExceptionHandler);
+            previousDefaultUncaughtExceptionHandler = null;
+            replacedDefaultUncaughtExceptionHandler = false;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/mongodb/native/blocking-checkpoint-storage/src/test/java/org/occurrent/subscription/mongodb/nativedriver/blocking/NativeMongoCheckpointStorageTest.java:139]: Good catch, and an embarrassing one to leave in, since a swallowing handler outliving its test is the exact failure this change exists to prevent. If either call threw, the restore never ran and every later test in that JVM would silently drop an `IllegalArgumentException` with that message.

I moved the restore into a `finally` around the shutdown and close.

File: subscription/mongodb/native/blocking-checkpoint-storage/src/test/java/org/occurrent/subscription/mongodb/nativedriver/blocking/NativeMongoCheckpointStorageTest.java [hunk 2]
```diff
@@ -121,8 +123,39 @@ void create_mongo_event_store() {
...
+        replacedDefaultUncaughtExceptionHandler = true;
+        Thread.setDefaultUncaughtExceptionHandler((thread, throwable) -> {
+            if (type.isInstance(throwable) && message.equals(throwable.getMessage())) {
+                return;
+            }
+            if (previous == null) {
+                thread.getThreadGroup().uncaughtException(thread, throwable);
+            } else {
+                previous.uncaughtException(thread, throwable);
+            }
```
[the remark this answers is not in the document]
Comment (johanhaleby) [reply subscription/mongodb/native/blocking-checkpoint-storage/src/test/java/org/occurrent/subscription/mongodb/nativedriver/blocking/NativeMongoCheckpointStorageTest.java:161]: You're right, and it is worse than an edge case. `ThreadGroup.uncaughtException` walks up to the root group and then asks `Thread.getDefaultUncaughtExceptionHandler` for a handler, which by then is this one, so it comes straight back. `Thread.getDefaultUncaughtExceptionHandler` is null in a plain JVM, so `previous == null` is the normal path here, not a rare one.

I reproduced it in a standalone JVM before changing anything. Delegating to the thread group turns an unexpected `IllegalStateException` into `Exception: java.lang.StackOverflowError thrown from the UncaughtExceptionHandler in thread "probe"`, so the real exception is destroyed by the handler that was supposed to report it. Printing directly gives `Exception in thread "probe" java.lang.IllegalStateException: unexpected, must be reported`.

Now it prints what the JVM would print when there is no previous handler.
